src/software/replay/replay_logger.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <experimental/filesystem>
+#include "software/multithreading/ordered_threaded_observer.h"
+#include "software/proto/sensor_msg.pb.h"
+#include "software/proto/tbots_replay.pb.h"
+
+
+class ReplayLogger : public OrderedThreadedObserver<SensorMsg>
+{
+   public:
+    explicit ReplayLogger(const std::string& output_directory,
+                          int _frames_per_chunk = DEFAULT_FRAMES_PER_CHUNK);
+    ReplayLogger(const ReplayLogger&) = delete;
+    ~ReplayLogger() override;
+   private:
+    void onValueReceived(SensorMsg frame) override;
+    void nextChunk();
```

Comment:
High-level question: What's the benefit of breaking things down into chunks? Seems like just logging everything to a single file is easier to implement, and easier to share around if needed. Do we collect so much data the file just gets way too large?

Reply:
The goal of this is to prevent too much data from being lost in the event of a crash

Reply:
Makes sense

Reply:
Would be good to add that as a comment somewhere, perhaps at the top of the logger class

Reply:
done